integreat_cms/cms/forms/regions/region_form.py

```diff
@@ -252,6 +255,8 @@ def save(self, commit=True):
             # Create links for the most recent versions of all translations manually
             find_links(region)
```

Comment:
Maybe it would make sense to execute this in the background as well?

Reply:
Let's see what the worst case scenario could be:
Source region A has a page /A/de/willkommen.
Further there is a target region B.

We start the thread find_links.
We start the thread replace_links.

find_links is creating links in the background, however /A/de/willkommen has not yet processed. Then /A/de/willkommen occurs in the content and replace_links updates the internal link to /B/de/willkommen. Later find_links creates the link /B/de/willkommen.
Everyone is happy =).

I'll also thread the find_links functionallity.

Reply:
Never mind. When running find_links in a thread, the links are not updated. Could it be that the save in find links overwrites the replaced links?

Reply:
Ah yes, `replace_links()` depends on `find_links()`. So maybe an intermediate function that runs them both synchronously, and just this intermediary function is started as one background thread?